### PR TITLE
Added some query filters on Job executions

### PR DESCRIPTION
**Pull request by MGSousa**

### CODE REVIEW

**Review by victorcoder — CHANGES_REQUESTED** [review]
LGTM but could you rename `JobExecutionsOptions` to `ExecutionOptions`?

**Review by victorcoder — CHANGES_REQUESTED** [review]
This won't work unless executions are stored in json in BuntDB, now they are stored in binary protobuf, or am I missing something?

**Comment by MGSousa** [comment]
Yes, the executions are stored in binary Protobuf with `proto.Marshal` but these changes will be ready for JSON marshaling in the future (if migrated), also it contains an option to order all executions by ASC or DESC and by itself is helpful.

**Comment by MGSousa** [comment]
> LGTM but could you rename `JobExecutionsOptions` to `ExecutionOptions`?

Done!

File: dkron/storage.go [hunk 1]
```diff
@@ -16,7 +16,7 @@ type Storage interface {
 	SetExecutionDone(execution *Execution) (bool, error)
 	GetJobs(options *JobOptions) ([]*Job, error)
 	GetJob(name string, options *JobOptions) (*Job, error)
-	GetExecutions(jobName string, timezone *time.Location) ([]*Execution, error)
+	GetExecutions(jobName string, timezone *time.Location, opts *ExecutionOptions) ([]*Execution, error)
```
Comment (victorcoder) [inline dkron/storage.go:19]: It would be great to refactor `timezone` into `ExecutionOptions`

**Review by victorcoder — CHANGES_REQUESTED** [review]

**Comment by victorcoder** [comment]
> Yes, the executions are stored in binary Protobuf with proto.Marshal but these changes will be ready for JSON marshaling in the future (if migrated), 

Then why implement this if it does not work? You can implement json marshal/unmarshal with fallback to proto.Unmarshal as I did with jobs, take a look at https://github.com/distribworks/dkron/blob/master/dkron/store.go#L309-L315

> also it contains an option to order all executions by ASC or DESC and by itself is helpful.

You mean ASC or DESC by the key?

**Comment by MGSousa** [comment]
> You mean ASC or DESC by the key?

Indeed.

File: dkron/storage.go (hunk 1 above)
Comment (MGSousa) [reply dkron/storage.go]: Done

**Review by victorcoder — APPROVED** [review]
LGTM

**Pull request merged**